File: backend/entity/maintenance.go
package entity

import (
    "time"
    "gorm.io/gorm"
)

type Maintenance struct {
    gorm.Model
    Title      string    `json:"title"`
    Detail     string    `json:"detail"`
    ReportDate time.Time `json:"report_date"`

    ImageURL  *string `json:"image_url"`
    ImageName *string `json:"image_name"`

    StudentID *uint   `json:"student_id"`
    Student   Student `gorm:"foreignKey:StudentID" json:"student"`

    RoomID *uint `json:"room_id"`
    Room   Room  `gorm:"foreignKey:RoomID" json:"room"`

    ProblemTypeID *uint       `json:"problem_type_id"`
    ProblemType   ProblemType `gorm:"foreignKey:ProblemTypeID" json:"problem_type"`

    MaintenanceStatusID *uint             `json:"maintenance_status_id"`
    MaintenanceStatus   MaintenanceStatus `gorm:"foreignKey:MaintenanceStatusID" json:"maintenance_status"`
}

File: frontend/src/interfaces/Maintenance.ts
import type { StudentInterface } from "./Student";
import type { ProblemTypeInterface } from "./ProblemType";
import type { MaintenanceStatusInterface } from "./MaintenanceStatus";
import type { RoomInterface } from "./Room";

export interface MaintenanceInterface {
  ID?: number;

  Title?: string;  title?: string;
  Detail?: string; detail?: string;
  ReportDate?: string | Date; report_date?: string;

  ImageURL?: string; image_url?: string;
  ImageName?: string; image_name?: string;

  StudentID?: number; student_id?: number;
  ProblemTypeID?: number; problem_type_id?: number;
  MaintenanceStatusID?: number; maintenance_status_id?: number;

  // ✅ snapshot ห้อง
  RoomID?: number; room_id?: number;

  // ส่งรูปแบบ base64 (สร้าง/แก้ไข)
  image_base64?: string;
  image_name?: string;

  Student?: StudentInterface;
  ProblemType?: ProblemTypeInterface;
  MaintenanceStatus?: MaintenanceStatusInterface;
  Room?: RoomInterface; room?: RoomInterface;
}
